lib/method/registerMethodFactory.js

```diff
@@ -47,90 +51,68 @@ function registerMethodFactory(dapiClient, dpp, wallet, blockchainIdentity) {
     ).toString('hex');
 
     // 1. Create preorder document
-    const preorderDocument = dpp.document.create('preorder', {
-      saltedDomainHash,
-    });
-
-    const prevSTHash = await dapiClient.getLastUserStateTransitionHash(
-      blockchainIdentity.regTxId,
+    const preorderDocument = dpp.document.create(
+      dataContract,
+      identity.getId(),
+      'preorder',
+      {
+        saltedDomainHash,
+      },
     );
 
     // 2. Create and send preorder state tranisition
-    const preorderSTPacket = dpp.packet.create([preorderDocument]);
-
-    const preorderTransaction = new Transaction()
-      .setType(Transaction.TYPES.TRANSACTION_SUBTX_TRANSITION);
+    const preorderTransition = dpp.document.createStateTransition([preorderDocument]);
+    preorderTransition.sign(identity.getPublicKeyById(0), privateKey);
 
-    preorderTransaction.extraPayload
-      .setRegTxId(blockchainIdentity.regTxId)
-      .setHashPrevSubTx(prevSTHash)
-      .setHashSTPacket(preorderSTPacket.hash())
-      .setCreditFee(1000)
-      .sign(privateKey);
-
-    const preorderValidationResult = dpp.packet.validate(preorderSTPacket);
-
-    if (!preorderValidationResult.isValid()) {
-      throw new Error('Preorder packet is not valid');
-    }
-
-    const preorderTransitionHash = await dapiClient.sendRawTransition(
-      preorderTransaction.serialize(),
-      preorderSTPacket.serialize().toString('hex'),
-    );
+    await dapiClient.updateState(preorderTransition);
 
     // 3. Wait until preorder state transition is confirmed
     await waitWithAttempts(10, async () => dapiClient
       .fetchDocuments(
-        dpp.getContract().getId(),
+        dataContract.getId(),
         'preorder',
-        { where: [['saltedDomainHash', '==', preorderDocument.getData().saltedDomainHash]] },
+        {
+          where: [[
+            'saltedDomainHash', '==', saltedDomainHash,
+          ]],
+        },
       ));
 
     // 4. Create domain document
     const nameHash = hash(Buffer.from(name)).toString('hex');
 
-    const domainDocument = dpp.document.create('domain', {
-      nameHash,
-      label,
-      normalizedLabel,
-      normalizedParentDomainName,
-      preorderSalt,
-      records: {
-        dashIdentity: blockchainIdentity.regTxId,
+    const domainDocument = dpp.document.create(
+      dataContract,
+      identity.getId(),
+      'domain',
+      {
+        nameHash,
+        label,
+        normalizedLabel,
+        normalizedParentDomainName,
+        preorderSalt,
+        records: {
```

Comment:
Records supposed to be an argument.
Maybe we should pass the user as well as an argument?